src/Return.jl

```diff
@@ -0,0 +1,113 @@
+"""
+    @match_fail
+
+This statement permits early-exit from the value of a @match case.
```

Comment:
is it worth calling out this applies to both `@match` and `@match2`? They are mostly the same, but maybe worth being super explicit in the docs

Reply:
Actually, it isn't wired to `@match`.  Fixing the doc.